src/v/cloud_storage/topic_manifest.cc

```diff
@@ -226,10 +226,17 @@ struct topic_manifest_handler
     };
 };
 
+// use feature table to decide if to encode with serde
 topic_manifest::topic_manifest(
-  const cluster::topic_configuration& cfg, model::initial_revision_id rev)
+  const cluster::topic_configuration& cfg,
+  model::initial_revision_id rev,
+  const features::feature_table& ft)
   : _topic_config(cfg)
-  , _rev(rev) {}
+  , _rev(rev)
+  , _manifest_version(
+      ft.is_active(features::feature::cluster_topic_manifest_format_v2)
+        ? serde_version
+        : first_version) {}
```

Comment:
is topic_manifest a sufficiently ephemeral type such that after upgrade completes, without brokers restarting, that this will take affect properly or will previous instantationed topic manifest structures be "stuck" on first_version?

Reply:
it's only local to ntp_archiver_service and goes out of scope quickly